src/GitVersion.Core/VersionCalculation/PathFilter.cs

```diff
@@ -0,0 +1,57 @@
+using GitVersion.Extensions;
+using GitVersion.Git;
+
+namespace GitVersion.VersionCalculation;
+
+internal class PathFilter(IGitRepository repository, GitVersionContext context, IEnumerable<string> paths, PathFilter.PathFilterMode mode = PathFilter.PathFilterMode.Inclusive) : IVersionFilter
+{
+    public enum PathFilterMode { Inclusive = 0, Exclusive = 1 }
+
+    private readonly IEnumerable<string> paths = paths.NotNull();
+    private readonly PathFilterMode mode = mode;
+    private readonly GitVersionContext context = context;
+
+    public bool Exclude(IBaseVersion baseVersion, out string? reason)
+    {
+        ArgumentNullException.ThrowIfNull(baseVersion);
+
+        reason = null;
+        if (baseVersion.SourceType != VersionIncrementSourceType.Tree) return false;
+
+        return Exclude(baseVersion.BaseVersionSource, out reason);
+    }
+
+    public bool Exclude(ICommit? commit, out string? reason)
+    {
+        commit.NotNull();
+        reason = null;
+
+        if (commit != null)
+        {
+            var patchPaths = repository.FindPatchPaths(commit, context.Configuration.TagPrefixPattern);
+
+            if (patchPaths != null)
+            {
+                switch (mode)
```

Comment:
Please prefix local field access with `this.`.

```suggestion
                switch (this.mode)
```